autometa/taxonomy/ncbi.py

```diff
@@ -167,117 +158,126 @@ def lineage(self, taxid, canonical=True):
-                    taxids_.update({taxid: {rank: name}})
+                if taxid not in ranked_taxids:
+                    ranked_taxids.update({taxid: {rank: name}})
                 else:
-                    taxids_[taxid].update({rank: name})
-        df = pd.DataFrame(taxids_).transpose()
+                    ranked_taxids[taxid].update({rank: name})
+        df = pd.DataFrame(ranked_taxids).transpose()
         df.index.name = 'taxid'
         if fillna:
             df.fillna(value='unclassified', inplace=True)
         return df
 
     def rank(self, taxid):
-        """Short summary.
+        """
+        Return the respective rank of provided `taxid`. If the taxid is deprecated, suppressed, 
+        withdrawn from NCBI (basically old) the updated rank will be retrieved
 
         Parameters
         ----------
         taxid : int
-            `taxid` to retrieve rank from `nodes`.
+            `taxid` to retrieve rank from `nodes.dmp`    
 
         Returns
         -------
         str
-            rank name
+            'rank name' if taxid is found in nodes.dmp and 'unclassified' if not 
 
         Raises
-        -------
+        ------
         ValueError
-            Provided `taxid` is not an integer
-
+            `taxid` is not valid
         """
-        # If taxid is not found in nodes.dmp, will return 'unclassified'
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
+        if not self.is_valid_taxid(taxid):
+            raise ValueError(
+                f'Taxid must be a positive integer! Given: {taxid}')
         return self.nodes.get(taxid, {'rank': 'unclassified'}).get('rank')
 
     def parent(self, taxid):
-        """Retrieve the parent taxid of provided taxid
+        """
+        Retrieve the parent taxid of provided `taxid`
 
         Parameters
         ----------
         taxid : int
-            Description of parameter `taxid`.
+            `taxid` whose parent `taxid` is being returned from `nodes.dmp`
 
         Returns
         -------
-        int
-            parent taxid
+        int or None
+            Parent `taxid` if found in `nodes.dmp` otherwise None
+
+        Raises
+        ------
+        ValueError
+            `taxid` is not valid
         """
-        # If taxid is not found in nodes.dmp, will return 1
-        try:
-            taxid = int(taxid)
-        except ValueError as err:
-            logger.error(
-                f'Taxid must be an integer! {taxid} type --> {type(taxid)}')
-            return None
-        return self.nodes.get(taxid, {'parent': 1}).get('parent')
+        if not self.is_valid_taxid(taxid):
+            raise ValueError(
+                f'Taxid must be a positive integer! Given: {taxid}')
+        return self.nodes.get(taxid, {'parent': None}).get('parent')
```

Comment:
You are returning `None` here instead of `1`. How will this affect other methods? 

To name a couple:

* `self.lineage`?
* `self.name(<non-existent taxid>, rank='genus')`?